### PR TITLE
Fix developer tooling for black, flake8

### PR DESCRIPTION
**Pull request by medmunds**
Several minor fixes and improvements related to addition
of black code formatter:

Makefile:
* Add `make install-tools` target to install development
  tooling needed for `make format` and `make lint`
* Add `make develop` target that installs tools and
  test package

README:
* Update Contributing section to use `make develop` and
  recommend `make format-fix`
* Note availability of tox for local testing on all Python
  versions

Travis.yml:
* Remove accidental duplicate `include:` that was preventing
  flake8 job from running
* Add names to flake8 and black jobs ("lint" and "format")
  to improve build reports

tox.ini:
* Add "format" tox env that runs black
* Change "lint" tox env to use make targets

setup.cfg:
* Update flake8 config with [recommended options for black][1]
  compatibility (but can't change max-line-length yet
  due to long doc strings)

[1]: https://github.com/psf/black/blob/master/docs/compatible_configs.md#flake8

### CODE REVIEW

**Comment by medmunds** [comment]
(I couldn't tell whether you had intended to keep flake8 alongside black, but guessed "yes" based on flake8 still appearing in the `make lint` target and an accidentally-disabled Travis job.)

File: tox.ini
Comment (paulmelnikow) [inline tox.ini:4]: Could you update this to match the matrix in the Travis config?

File: .travis.yml
Comment (paulmelnikow) [inline .travis.yml:21]: Thanks for fixing this. I have the hardest time making sense of Travis' config format.

**Review by paulmelnikow — COMMENTED** [review]
Thanks for picking this up!

File: .travis.yml
Comment (paulmelnikow) [inline .travis.yml:22]: This can be removed now.

**Comment by paulmelnikow** [comment]
>     * Install black in `make install`, so it is available
>       for `make format`
> 
>     * Restore flake8 to `make install`, so it is available
>       for `make lint`

Thanks for addressing this in particular. It wasn't possible until #133 was merged and was on my mind to address.

File: .travis.yml
Comment (medmunds) [reply .travis.yml:22]: Sorry, not following what you want to remove. Do you want Travis to run:

A. Two separate code-checking jobs: "lint" (flake8) and "format" (black)?
B. Only one code-checking job that includes both flake8 and black?
C. Neither?

A is what's in this PR ([build output](https://travis-ci.org/github/syrusakbary/snapshottest/builds/731710336))

File: .travis.yml
Comment (medmunds) [reply .travis.yml:21]: Yep. (I'm in the process of switching over to GitHub actions in some of my other projects.)

File: .travis.yml
Comment (paulmelnikow) [reply .travis.yml:22]: We don't need to run `pip install black==20.8b1` here because you've added it to `make install`.

It had been installed here, instead of through `make install`, because black==2.0.8b1 can't be installed under Python 2.7.

**Comment by paulmelnikow** [comment]
> (I couldn't tell whether you had intended to keep flake8 alongside black, but guessed "yes" based on flake8 still appearing in the `make lint` target and an accidentally-disabled Travis job.)

Yes, definitely. 😁 They do complementary things.

I usually ignore `E203, E302, W503, I201` in the flake8 config to avoid conflicts with black. Not sure if those should be added here at this time.

**Comment by medmunds** [comment]
Oh, OK, I see a problem. Travis is using `make install`—but with this PR that would include the developer tooling in the tests, which we don't really want. (Plus it fails on Python 3.5, because black is no longer available there.)

Two options:

* Change Travis's install step to not use `make install`, but instead just run `pip install -e ".[test]"`
* Put `make install` back the way it was, and add a new `make develop` target that also installs black and flake8 (and update the README Contributing section)

Do you have a preference?

**Comment by medmunds** [comment]
> I usually ignore E203, E302, W503, I201 in the flake8 config to avoid conflicts with black. Not sure if those should be added here at this time.

I just copied the (current) [recommended flake8 settings in the black docs](https://github.com/psf/black/blob/master/docs/compatible_configs.md#flake8). Easy enough to add the others if it would be helpful.

File: .travis.yml
Comment (medmunds) [reply .travis.yml:22]: Got it. But, um, see [my downstream comment](https://github.com/syrusakbary/snapshottest/pull/135#issuecomment-701618483). We still need to split apart linting/formatting install from testing install, one way or another.

**Comment by paulmelnikow** [comment]
> * Put `make install` back the way it was, and add a new `make develop` target that also installs black and flake8 (and update the README Contributing section)

This seems like a good approach to me.

**Comment by paulmelnikow** [comment]
> I just copied the (current) [recommended flake8 settings in the black docs](https://github.com/psf/black/blob/master/docs/compatible_configs.md#flake8). Easy enough to add the others if it would be helpful.

I wonder if W503 is no longer a conflict.

I think E302 and I201 could be left off, too. Those are just my own style preferences.

File: tox.ini
Comment (paulmelnikow) [reply tox.ini:4]: Addressed in #136.

File: .travis.yml
Comment (paulmelnikow) [reply .travis.yml:21]: Yea, we should discuss which CI we want to use. I've had mixed experiences with GH Actions. They haven't been as reliable as other CI's I've used. Though I like that they are easy to use on other people's projects. 😁 My default these days is still CircleCI, though it has its own downsides.

That Travis builds aren't showing up in the PRs is a bit aggravating. 😭

**Comment by medmunds** [comment]
Sounds like W503 is now disabled by default in flake8, since a PEP 8 change a while back.

I think I201 depends on flake8-import-order, which we aren't currently using. (I don't think that's included with flake8 by default.)

I have no opinion on E302. (Well, actually, I agree with you, but in the spirit of deferring to opinionated formatting tools, I'm keeping my own opinion to myself :grin:)

File: .travis.yml
Comment (medmunds) [reply .travis.yml:21]: About once a year, something mysteriously stops working with Travis-CI integration, and I end up having to disconnect and reconnect it to my repos. (To be fair, I think this is partly GH's fault, as they've changed how tools are supposed to integrate over the years.)

**Comment by paulmelnikow** [comment]
Good to know that I can remove W503 from other projects. Yea, let's add flake8-import-order at some point, and then decide how we feel about all the newlines. I feel like without E302 the imports take up a whole screenful. 😁

**Review by paulmelnikow — APPROVED** [review]

**Comment by paulmelnikow** [comment]
Ready to merge, right?

**Comment by medmunds** [comment]
Yep!

**Comment by paulmelnikow** [comment]
Thanks so much! 😁 🚀

**Comment by medmunds** [comment]
@paulmelnikow my pleasure. Thank *you* for jumping in as an active maintainer!

**Pull request merged**
